common/src/main/java/com/wynntils/services/mapdata/type/MapArea.java

```diff
@@ -5,17 +5,21 @@
 package com.wynntils.services.mapdata.type;
 
 import com.wynntils.utils.mc.type.Location;
+import com.wynntils.utils.type.BoundingPolygon;
 import com.wynntils.utils.type.BoundingShape;
 import java.util.List;
 
 public interface MapArea extends MapFeature {
     // The area is described by a polygon. This list is the sequence of
-    // vertices of that polygon, ordered in a counterclockwise orientation.
+    // vertices of that convex polygon, ordered in a counterclockwise orientation.
```

Comment:
Is it necessary to limit this to convex polygons? I can think of many reasons for having non-convex polygons, e.g. if trying to create areas that cover geographical features on the Wynncraft map.

Reply:
This is very elaborate, the reasoning is that some operations are much more reasonable if they are only done on complex polygons, with these operations being rendering, and different kinds of clipping checks, both of which are vital performance wise.

Reply:
Okay. I'm honestly not very good at that kind of algorithms (I remember that I did have exams on stuff like convex hulls etc in uni, but I've forgotten all details...).

Maybe we can rephrase the text (which basically is our specification) to something like:
```
    // The area is described by a polygon. This list is the sequence of
    // vertices of that polygon, ordered in a counterclockwise orientation.
    // Currently, only convex polygons are supported.
```

to at least indicate that we could be open for something else, if the need arises.

Reply:
I still think this applies.

Reply:
I do not think that we should be open for something else. Concave polygons can be described as multiple convex polygons, and the complexity drastically changes when you introduce concave polygons into the mix.